packages/core/src/agent/Agent.ts

```diff
@@ -194,13 +194,13 @@ export class Agent {
       transport.stop()
     }
 
-    // close/delete wallet if still initialized
+    // close wallet if still initialized
     if (this.wallet.isInitialized) {
-      if (deleteWallet) {
-        await this.wallet.delete()
-      } else {
-        await this.wallet.close()
-      }
+      await this.wallet.close()
+    }
+    // delete wallet if it has been created
+    if (this.wallet.isCreated && deleteWallet) {
+      await this.wallet.delete()
     }
```

Comment:
There was a problem when I created a wallet but didn't call open then the shutdown method didn't delete the wallet.